alpha/apps/kaltura/lib/clipconcat/kClipManager.php

```diff
@@ -335,18 +362,42 @@ private function handleClipConcatParentJob($batchJob)
 				}
 				else
 				{
-					//start child clip jobs
-					$errDesc = '';
-					$this->addClipJobs($batchJob, $jobData->getTempEntryId(), $errDesc,
-						$jobData->getPartnerId(),
-						$jobData->getOperationAttributes(), kConvertJobData::TRIMMING_FLAVOR_PRIORITY);
-					kJobsManager::updateBatchJob($batchJob, BatchJob::BATCHJOB_STATUS_ALMOST_DONE);
+					$this->addClipJobsFromBatchJob($batchJob, $jobData);
 				}
 				break;
 			default:
 				break;
 		}
 	}
+	
+	/**
+	 * @param $batchJob
+	 * @param $jobData
+	 * @throws APIException
+	 * @throws kCoreException
+	 * Prepare Extract Media jobs for additional flavors if exist, and run the clipping
+	 */
+	private function addClipJobsFromBatchJob($batchJob, $jobData)
+	{
+		$flavorAssetsToBeProcessed = assetPeer::retrieveAudioFlavorsByEntryID($jobData->getSourceEntryId());
+		$originalFlavorAsset = assetPeer::retrieveOriginalByEntryId($jobData->getTempEntryId());
+		array_unshift($flavorAssetsToBeProcessed, $originalFlavorAsset);
+		foreach($flavorAssetsToBeProcessed as $asset)
+		{
+			/** @var flavorAsset $asset */
+			$this -> addClipJobsOnFlavor($batchJob, $jobData, $asset -> getId());
+		}
+		kJobsManager ::updateBatchJob($batchJob, BatchJob::BATCHJOB_STATUS_ALMOST_DONE);
+	}
+	
+	//function that runs addClipJobs on a flavor
+	private function addClipJobsOnFlavor($batchJob, $jobData, $assetId)
+	{
+		//start child clip jobs
+		$errDesc = '';
+		$this -> addClipJobs($batchJob, $jobData -> getTempEntryId(), $errDesc, $jobData -> getPartnerId(),
```

Comment:
since this function has no added value just calling addClipJobs please remove it